src/routers/controllers/addPresenterController.ts

```diff
@@ -2,11 +2,39 @@ import { Request, Response } from "express";
 import * as constants from "../../constants";
 import { getUrlWithCompanyNumber } from "../../lib/utils/urlUtils";
 import { AddPresenterHandler } from "../handlers/yourCompanies/addPresenterHandler";
+import { deleteExtraData, getExtraData } from "../../lib/utils/sessionUtils";
+import { redirectPage } from "../../lib/utils/referrerUtils";
 
 export const addPresenterControllerGet = async (req: Request, res: Response): Promise<void> => {
-    const handler = new AddPresenterHandler();
-    const viewData = await handler.execute(req, constants.GET);
-    res.render(constants.ADD_PRESENTER_PAGE, viewData);
+    const referrer: string | undefined = req.get("Referrer");
```

Comment:
Would it be worth moving this checking session data and referrers logic outside of the controllers themselves, in separate functions or did you try that at all? Meaning the navigation checking code can be the same but independent of the controllers.